PDP11/pdp11_rr.c

```diff
@@ -1200,21 +1315,21 @@ static const uint16 rr_boot_rom[] = {
  * R0     = UNIT NUMBER                                                                                *
  * R1     = CONTROLLER CSR                                                                             *
  * R2, R3 = TEMPORARIES                                                                                *
- * R4     = ALWAYS POINTS TO PROM BASE + 20 (HELPS LOCATE BOOTED DEVICE DESIGNATION)                   *
+ * R4     = ALWAYS POINTS TO PROM BASE + 20 (HELPS LOCATE THE BOOTED DEVICE DESIGNATION)               *
  * R5     = LAST COMMAND DATA (E.G. LOAD ADDR, EXAM DATA; OTHERWISE, JUNK)                             *
- * R6(SP) = PC OF THE COMMAND START (IN M9312 POINTS TO WHERE THE BOOT COMMAND ORIGINATED FROM)        */
+ * R6(SP) = PC OF THE COMMAND START (CONTAINS THE ADDRESS WHERE THE BOOT COMMAND ORIGINATED FROM)      */
 /*                                              .TITLE RP11 BOOT M9312 STYLE - TONY LAWRENCE (C) 2023  */
 /*                                              .ASECT                                                 */
 /* 002000                                       .=2000                                                 */
-/* 002000 */ 0042120,                 /* START: .WORD   "PD             ; "DP" (DEVICE DESIGNATION)    */
+/* 002000 */ 0042120,                 /* START: .WORD   "PD             ; "DP" (RP DEVICE DESIGNATION) */
 /* 002002 */ 0012706, BOOT_ENTRY,     /* BOOT:  MOV     #BOOT, SP       ; ENTRY POINT PC               */
 /* 002006 */ 0112700, 0000000,        /*        MOVB    #0, R0          ; UNIT NUMBER                  */
 /* 002012 */ 0012701, 0176726,        /*        MOV     #176726, R1     ; RPCS + 12                    */
-/* 002016 */ 0012704, BOOT_START+020, /*        MOV     #<START+20>, R4 ; BACKLINK TO ROM W/OFFSET 20  */
+/* 002016 */ 0012704, BOOT_START+020, /*        MOV     #<START+20>, R4 ; BACKLINK TO PROM W/OFFSET 20 */
 /* 002022 */ 0005041,                 /*        CLR     -(R1)           ; DISK ADDRESS                 */
 /* 002024 */ 0005041,                 /*        CLR     -(R1)           ; CYLINDER ADDRESS             */
 /* 002026 */ 0005041,                 /*        CLR     -(R1)           ; MEMORY ADDRESS               */
-/* 002030 */ 0012741, 0177000,        /*        MOV     #-512., -(R1)   ; WORD COUNT                   */
+/* 002030 */ 0012741, 0177000,        /*        MOV     #-512., -(R1)   ; WORD CT (ALWAYS 2 FULL SECTS)*/

```

Comment:
Two sectors?  That's odd, I thought that bootstraps always read 512 bytes (one sector, in most disks).

Reply:
Paul, please read listings for M9312 -- for disk devices the word count is always 2 full sectors.

Reply:
May I ask why are you scrutinizing my code comments?

Reply:
In this particular case, while the question was prompted by what I read in the comments it is actually about the substance of the code.
I'll definitely have to take a look at the M9312, I wonder if that's a mistake or an intentional change introduced in that device.  In the older M792 boot card (the one with discrete diodes), the word count is definitely -256, not -512.  In the MR-11B boot it's -512, though.  I wonder if that's because it added support for the TM11, which has a byte count rather than a word count.  So -512 there means 512 bytes, the standard boot block size.

Reply:
> I'll definitely have to take a look at the M9312

To make your investigation easier, you can take a look at M9312 listings here (among other places, obviously):

http://www.ak6dn.com/PDP-11/M9312/

RP11 bootstrap, and RK11 (and most everything else what I checked in there for disks) are doing `-512.` for the WC registers.  Whether was it a bug or not, is no longer relevant now, I suppose, because it has been like that for all this time.  IIRC, RSTS keeps time in the _last_ 3 tops words of the "first sector" (loc octal 772, 774, and 776), when it is doing a software bootstrap (or reboot).  In case of hardware bootstrap (that we're talking about here), those locations are obviously overwritten by the boot sector (regardless of whether only one sector is getting loaded, or two).

Reply:
Yes, good point.  Thanks for the link.